tests/unit/conftest.py

```diff
@@ -120,6 +120,40 @@ def mod_tmp_run_dir(tmp_path_factory: pytest.TempPathFactory):
         return tmp_run_dir(tmp_path, mp)
 
 
+def tmp_src_dir(tmp_path: Path):
+    """Fixture that creates a temporary workflow source dir.
+
+    (Actually the fixture is below, this is the re-usable meat of it.)
+
+    Args:
+        path: Path of source dir relative to cylc-src/.
+
+    Example:
+        src_dir = tmp_src_dir('foo')
+    """
+    def _tmp_src_dir(path: Union[Path, str]) -> Path:
+        cylc_src_dir = tmp_path / 'cylc-src'
+        cylc_src_dir.mkdir(exist_ok=True)
+        src_dir = cylc_src_dir / path
+        src_dir.mkdir(parents=True)
+        (src_dir / WorkflowFiles.FLOW_FILE).touch()
+        return src_dir
+    return _tmp_src_dir
+
+
+@pytest.fixture(name='tmp_src_dir')
+def tmp_src_dir_fixture(tmp_path: Path):
+    # This is the actual tmp_src_dir fixture
+    return tmp_src_dir(tmp_path)
+
+
+@pytest.fixture(scope='module')
+def mod_tmp_src_dir(tmp_path_factory: pytest.TempPathFactory):
+    """Module-scoped version of tmp_src_dir()"""
+    tmp_path = tmp_path_factory.getbasetemp()
+    return tmp_src_dir(tmp_path)
+
+
```

Comment:
I think there's a better way of doing this following the pattern used for `run_dir`. Can plug in the `flow()` stuff too:

```suggestion
# copying the patter used for run_dir

@pytest.fixture(scope='session')
def ses_src_dir():
    ...
    
@pytest.fixture(scope='module')
def mod_src_dir(ses_src_dir, request):
    return srs_src_dir / request.module.__name__
    
@pytest.fixture
def src_dir(mod_src_dir, request):
    return mod_src_dir / request.function.__name__
    
# integrating the flow stuff

@pytest.fixture(scope='module')
def mod_src_flow(mod_src_dir):
    yield partial(_make_flow, mod_src_dir, mod_src_dir)
    
@pytest.fixture
def src_flow(mod_src_dir):
    yield partial(_make_flow, src_dir, src_dir)
```